src/coreclr/vm/ceeload.cpp

```diff
@@ -11901,6 +11901,8 @@ HRESULT Module::WriteMethodProfileDataLogFile(bool cleanup)
                 _ASSERTE(!"Error writing ibc profile data to file");
                 hr = HRESULT_FROM_WIN32(lasterror);
             }
+
+            delete pEmitter;
```

Comment:
The VM convention is to use `NewHolder<ProfileEmitter>` for cleanup in situations like this one. Could you please switch to that?